src/actions/keyBasedAssetTransferActions.js

```diff
@@ -0,0 +1,367 @@
+  const { nonce: lastNonce, transactionCount: lastCount } = signedTransaction;
+  const txCountNew = { lastCount, lastNonce };
+  dispatch({ type: UPDATE_TX_COUNT, payload: txCountNew });
+  dispatch(saveDbAction('txCount', { txCount: txCountNew }, true));
+
+  return signedTransaction;
+};
+
+export const removeKeyBasedAssetToTransferAction = (assetData: AssetData) => {
+  return (dispatch: Dispatch, getState: GetState) => {
+    const { keyBasedAssetTransfer: { data: keyBasedAssetsToTransfer } } = getState();
+
+    // filter out matching
+    const updatedKeyBasedAssetsToTransfer = keyBasedAssetsToTransfer.filter(({ assetData: transferAssetData }) => {
+      if (transferAssetData?.tokenType !== COLLECTIBLES) return transferAssetData?.token !== assetData.token;
+      const isMatchingCollectible = transferAssetData?.id === assetData?.id
+        && addressesEqual(transferAssetData?.contractAddress, assetData?.contractAddress);
+      return !isMatchingCollectible;
+    });
+
+    dispatch({ type: SET_KEY_BASED_ASSETS_TO_TRANSFER, payload: updatedKeyBasedAssetsToTransfer });
+  };
+};
+
+export const addKeyBasedAssetToTransferAction = (assetData: AssetData, amount?: number) => {
+  return (dispatch: Dispatch, getState: GetState) => {
+    const { keyBasedAssetTransfer: { data: keyBasedAssetsToTransfer } } = getState();
+    const updatedKeyBasedAssetsToTransfer = keyBasedAssetsToTransfer.concat({ assetData, amount });
+    dispatch({ type: SET_KEY_BASED_ASSETS_TO_TRANSFER, payload: updatedKeyBasedAssetsToTransfer });
+  };
+};
+
+export const fetchAvailableBalancesToTransferAction = () => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    const {
+      wallet: { data: { address: keyBasedWalletAddress } },
+      assets: { supportedAssets },
+    } = getState();
+    dispatch({ type: SET_FETCHING_AVAILABLE_KEY_BASED_BALANCES_TO_TRANSFER });
+
+    // fetch key based assets
+    const ownedAssets = await getAllOwnedAssets(api, keyBasedWalletAddress, supportedAssets);
+    const availableBalances = await api.fetchBalances({
+      address: keyBasedWalletAddress,
+      assets: getAssetsAsList(ownedAssets),
+    });
+
+    dispatch({
+      type: SET_AVAILABLE_KEY_BASED_BALANCES_TO_TRANSFER,
+      payload: transformBalancesToObject(availableBalances),
+    });
+  };
+};
+
+export const fetchAvailableCollectiblesToTransferAction = () => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    const { wallet: { data: { address: keyBasedWalletAddress } } } = getState();
+
+    dispatch({ type: SET_FETCHING_AVAILABLE_KEY_BASED_COLLECTIBLES_TO_TRANSFER });
+
+    let availableCollectibles = [];
+
+    const fetchedCollectibles = await api.fetchCollectibles(keyBasedWalletAddress);
+    if (fetchedCollectibles.error || !fetchedCollectibles.assets) {
+      reportLog('Failed to fetch key based wallet collectibles', { requestResult: fetchedCollectibles });
```

Comment:
He won't lose anything if there would be an error in request – it won't appear in his UI and least what we can do is report our logs to Sentry (added already). However, I might need to remove `!fetchedCollectibles.assets` because it will always report this error if there's no collectibles for user even though it wasn't actually an error.

Reply:
Thanks for that clarification :)